### PR TITLE
Added basic lifecycle extension

### PR DESCRIPTION
**Pull request by DanteMustCode**
* updated `main.go` based on [template](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/extension/main.go) with a few changes:
  * added upgradev1beta1 in init()
  * removed infrav1 in init()
  * filled setupReconcilers
  * simplified setupLifecycleHookHandlers
  * removed setupTopologyMutationHookHandlers
  * removed (empty) setupIndexes
  * removed (empty) setupWebhooks
  * kept a few parameters (different from kubebuilder):
    * var metricsAddr + secureMetrics ~ diagnosticsOpts
    * var probeAddr ~ healthAddr
    * var enableHTTP2 ~ TBD
* removed `UpgradePolicy` controller
* renamed `UpgradeTaskPhase` const
* updated stub based on latest [proposal](https://hackmd.io/Wv_u2xXJQsaj4wWFQ3PqCQ)
* fix `repo` as `microsoft/cluster-api-inplace-upgrader`
* added generated files from `make manifests`

### CODE REVIEW

File: api/v1beta1/upgradetask_types.go [hunk 1]
```diff
@@ -19,32 +19,27 @@ package v1beta1
...
-	// Template *MachineUpgraderTemplate `json:"template,omitempty"`
-	Phase UpgradeTaskPhase `json:"phase,omitempty"`
+	NewMachineSpec         stub.MachineSpec         `json:"newMachineSpec,omitempty"`
 }
 
 // UpgradeTaskStatus defines the observed state of UpgradeTask
 type UpgradeTaskStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Phase UpgradeTaskPhase `json:"phase,omitempty"`
```
Comment (mogliang) [inline api/v1beta1/upgradetask_types.go:39]: any field in status shall only be updated by  cr reconciler. But we need set this phase to abort if we need abort a ongoing upgradeTask.

File: api/v1beta1/upgradetaskphase_types.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+type UpgradeTaskPhase string
+
+const (
```
Comment (mogliang) [inline api/v1beta1/upgradetaskphase_types.go:21]: same as above comment, we need a setting in spec to tell which target state we want to drive to

File: api/v1beta1/upgradetask_types.go (hunk 1 above)
Comment (DanteMustCode) [reply api/v1beta1/upgradetask_types.go:39]: `TargetPhase` is added to address this comment.

File: api/v1beta1/upgradetaskphase_types.go (hunk 2 above)
Comment (DanteMustCode) [reply api/v1beta1/upgradetaskphase_types.go:21]: `TargetPhase` is added to address this comment. To keep it simple, the same type is reused.

**Review by mogliang — APPROVED** [review]

**Pull request merged**